pandas/tests/groupby/test_groupby.py

```diff
@@ -1196,6 +1196,21 @@ def test_groupby_complex():
     tm.assert_series_equal(result, expected)
 
 
+def test_groupby_complex_2():
+    df = DataFrame(
+        [
+            {"a": 2, "b": 1 + 2j},
+            {"a": 1, "b": 1 + 1j},
+            {"a": 1, "b": 1 + 2j},
+        ]
+    )
+    assert df.groupby("b").groups == {(1 + 1j): [1], (1 + 2j): [0, 2]}
```

Comment:
You don't have to assert the groups, asserting the final result like you do below is sufficient